Mathlib/Data/Real/Sqrt.lean

```diff
@@ -478,6 +485,14 @@ instance : StarOrderedRing ℝ :=
 
 end Real
 
+instance NNReal.instStarOrderedRing : StarOrderedRing ℝ≥0 := by
+  refine .ofLEIff fun x y ↦ ⟨fun h ↦ ?_, ?_⟩
+  · obtain ⟨d, rfl⟩ := exists_add_of_le h
+    refine ⟨sqrt d, ?_⟩
+    simp only [star_trivial, mul_self_sqrt]
+  · rintro ⟨p, -, rfl⟩
+    exact le_self_add
```

Comment:
Are we missing some construction to pull star-ordered-rings along a strictly monotone function, or whatever the actual constraints are? This proof was pretty short, but for `NNRat` we have a bit more work cut out for us, and it would be nice to have the generalization.